### ISSUE TITLE
"Pipe All Records" button not working

### ISSUE THREAD

**Issue by nsonanh** [vanderbilt-redcap/cross-project-piping-module#21]
Hi, I'm having an issue with piping records using the `Pipe All Records` button in `Record Status Dashboard` page.

Now for simplicity let's say I have `project A` and `project B`.
- In project B I set up cross-piping so that once a `speciment_id` field is saved, other fields will be piped from project A.
- I created new record in `project B` with a `speciment_id` that exists in `project A`.
- I go to `Record Status Dashboard` and click `Pipe All Records`.
- The log result is in this screenshot: 
![image](https://user-images.githubusercontent.com/15029492/199055928-f504504c-a8c1-446b-88a4-9905609043ac.png)

Apparently no field was piped. If I open the form in `project B` I can see that fields can be piped when the form loads. However I'd need them to be automatically piped when I click the button. 

Does anyone have any advice on how I can do it ? Any help would be greatly appreciated.
Thanks!

**Comment by nsonanh** [comment]
I'm using version [v1.4.10](https://github.com/vanderbilt-redcap/cross-project-piping-module/tree/v1.4.10) btw

**Comment by WangiMw** [comment]
Hi, I am having the exact issue and I can not seem to find the solution. I hope someone will come to the rescue soon

**Comment by mmcev106** [comment]
We unfortunately haven't had a team willing to offer a general support budget for this module in years, which a requirement before we can log time on something.  @tertek, I might loop you in since you've shown interested in this module recently.

**Comment by nsonanh** [comment]
FYI, the issue lies in line `956` of file `CrossprojectpipingExternalModule.php` ([here](https://github.com/vanderbilt-redcap/cross-project-piping-module/blob/master/CrossprojectpipingExternalModule.php#L956)). Somehow the module failed this check while it runs.

I'm not sure I understand about this `event` check but if I updated the check to do nothing then the pipe-all button works correctly.

**Comment by WangiMw** [comment]
> FYI, the issue appears to be on line `956` of file `CrossprojectpipingExternalModule.php` ([here](https://github.com/vanderbilt-redcap/cross-project-piping-module/blob/master/CrossprojectpipingExternalModule.php#L956)). Somehow the module failed this check while it runs. If I updated the check to do nothing then the pipe-all button works correctly.

Did you just comment it out?

**Comment by nsonanh** [comment]
> Did you just comment it out?

If you comment it out the EM will throw an error. So I update it to do nothing, like this:
```
if (in_array($eid, (array) $source_project['valid_match_event_ids']) === false) {}
```

Apparently the check has a purpose, I'm still studying it to understand the rationale behind.

**Comment by nsonanh** [comment]
I attempt to fix in PR #23. @mmcev106 please see if we can include it in.

**Comment by mmcev106** [comment]
@nsonanh's PR has been merged & included in version 1.4.12!  I suspect there are still cases where the button does not work properly, but at least we fixed a few of them.  I'll close this for now.

**Comment by harpsicord86** [comment]
Hi @mmcev106 and @nsonanh,

Unfortunately, I seem to be having the same issue... Pipe All Records button seems to do nothing... I've not got any repeating instruments in my destination project but the auto-piping/piping button in record seems to work fine. I'm on v1.4.18 - REDCap version 13.2.3

**Comment by nsonanh** [comment]
Hi @harpsicord86,
Could you be a bit more specific on the setup of your projects e.g. ?

Upon testing again I can see that the pipe-all button does work (at least for normal records):

![image](https://user-images.githubusercontent.com/15029492/216352015-c2e8df77-3688-4220-9ef7-4516ab44cfae.png)

I'm using v1.4.18 - REDCap version 12.0.7

**Comment by nsonanh** [comment]
Hi @harpsicord86
It's strange that I tried to have a similar set up to yours and it's still working for me. I also updated my Redcap version to 13.2.3 too.
I'll provide some screenshots of my module config values and records for you:
![image](https://user-images.githubusercontent.com/15029492/216779754-1b095054-10c2-4a11-975c-5a2c3e7e39f3.png)
![image](https://user-images.githubusercontent.com/15029492/216779763-aad8e51f-ba08-441e-a985-f609fe951b0d.png)
![image](https://user-images.githubusercontent.com/15029492/216779838-ffdf1ebf-fc25-4e36-bdc3-14a912d94812.png)

**Comment by harpsicord86** [comment]
Hi @nsonanh 
I believe I may have a working theory as to why my cross-project piping is not working... I think it is because I'm trying to get data from within a separate event. I've only realised this after trying to attempt to copy data from a consent form between projects and the consent form is actually part of another event... could this be a potential bug?

**Comment by mmcev106** [comment]
@harpsicord86, that very well could be a bug.  I'm not sure @moorejr5 or anyone else would have time approved to look it, but I'd love to keep track of it for whenever someone is able to rise to the occasion.  If/when you're confident in your testing, would you mind creating a new issue describing the scenario?
